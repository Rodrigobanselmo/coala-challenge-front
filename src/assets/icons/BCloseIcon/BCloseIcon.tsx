import React, { FC } from "react";

import CloseOutlinedIcon from "@mui/icons-material/CloseOutlined";
import { IIconMuiProps } from "@/core/interfaces/IIconMuiProps";

export const BCloseIcon: FC<IIconMuiProps> = ({ ...props }) => {
  return <CloseOutlinedIcon {...props} />;
};

export default CloseOutlinedIcon;
