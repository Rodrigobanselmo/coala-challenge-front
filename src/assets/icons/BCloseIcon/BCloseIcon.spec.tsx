import React from "react";
import { render, screen } from "@testing-library/react";
import BCloseIcon from "./BCloseIcon";

describe("BCloseIcon", () => {
  it("renders without errors", () => {
    render(<BCloseIcon />);
    const icon = screen.getByTestId("CloseOutlinedIcon");

    expect(icon).toBeInTheDocument();
    expect(icon).toMatchSnapshot();
  });

  it("applies custom props", () => {
    render(<BCloseIcon fontSize="large" color="primary" />);

    const icon = screen.getByTestId("CloseOutlinedIcon");
    expect(icon).toMatchSnapshot();
  });
});
