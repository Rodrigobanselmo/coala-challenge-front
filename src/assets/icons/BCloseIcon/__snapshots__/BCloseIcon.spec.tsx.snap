// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BCloseIcon applies custom props 1`] = `
<svg
  aria-hidden="true"
  class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeLarge css-bsqp9w-MuiSvgIcon-root"
  data-testid="CloseOutlinedIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
  />
</svg>
`;

exports[`BCloseIcon renders without errors 1`] = `
<svg
  aria-hidden="true"
  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
  data-testid="CloseOutlinedIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
  />
</svg>
`;
